addons/pos_sale/tests/test_pos_sale_flow.py

```diff
@@ -123,3 +123,50 @@ def test_settle_order_with_different_product(self):
         self.assertEqual(orderline_product_a.move_ids.product_uom_qty, 0)
         # 1 item to deliver for product b.
         self.assertEqual(orderline_product_b.move_ids.product_uom_qty, 1)
+
+    def test_settle_order_with_promotions(self):
+        if not self.env["ir.module.module"].search([("name", "=", "pos_coupon"), ("state", "=", "installed")]):
+            self.skipTest("pos_coupon module is required for this test")
```

Comment:
I added this so the test doesn't run when the required module are not installed.